File: 2025-1/sesion-de-clase-18/Pedido.cpp
/* 
 * File:   Pedido.cpp
 * Author: erichuiza
 * 
 * Created on June 2, 2025, 1:19 PM
 */

#include "Pedido.h"

Pedido::Pedido() {
    this->codigo = nullptr;
    this->plato = nullptr;
}

Pedido::Pedido(const char* codigo, int clienteDni, const char* plato) : Registro() {
    this->setCodigo(codigo);
    this->setClienteDni(clienteDni);
    this->setPlato(plato);
} 

Pedido::Pedido(const Pedido& orig) : Registro(orig) {
    *this = orig;
}

int Pedido::getClienteDni() const {
    return clienteDni;
}

void Pedido::setClienteDni(int clienteDni) {
    this->clienteDni = clienteDni;
}

char* Pedido::getCodigo() const {
    return codigo;
}

void Pedido::setCodigo(const char* codigo) {
    if (this->codigo) {
        delete[] this->codigo;
    }
    
    this->codigo = new char[strlen(codigo) + 1];
    strcpy(this->codigo, codigo);
}

char* Pedido::getPlato() const {
    return plato;
}

void Pedido::setPlato(const char* plato) {
    if (this->plato) {
        delete[] this->plato;
    }
    
    this->plato = new char[strlen(plato) + 1];
    strcpy(this->plato, plato);
}

bool Pedido::cargarDatos(ifstream& archivo) {
    if (!archivo.eof()) {
        char* codigo = leerCadena(archivo);
        int clienteDni;
        archivo >> clienteDni;
        
        archivo.ignore();
        char* plato = leerCadena(archivo, '\n');
        
        this->setCodigo(codigo);
        this->setClienteDni(clienteDni);
        this->setPlato(plato);
        
        return true;
    }
    
    return false;
}

Pedido& Pedido::operator=(const Pedido& pedido) {
    this->setCodigo(pedido.getCodigo());
    this->setClienteDni(pedido.getClienteDni());
    this->setPlato(pedido.getPlato());
    
    return *this;
}

void Pedido::imprimir(ostream& os) const {
    os 
        << left << setw(10) << (this->getPrioridad() == PedidoPrioridad::ALTA ? "Si" : "No")
        << setw(15) << this->codigo
        << setw(15) << this->clienteDni
        << setw(50) << this->plato
        << endl;
}

Pedido::~Pedido() {
    delete[] codigo;
    delete[] plato;
}

File: 2025-1/sesion-de-clase-24-1/src/Pedido.java

import java.util.Scanner;

/**
 *
 * @author erichuiza
 */
public abstract class Pedido extends Registro {
    private String codigo;
    private int clienteDni;
    private String plato;
    
    public Pedido() {
    }
    
    public Pedido(String codigo, int clienteDni, String plato) {
        this.codigo = codigo;
        this.clienteDni = clienteDni;
        this.plato = plato;
    }
    
    public abstract int getTipo();
    
    public String getCodigo() {
        return codigo;
    }

    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }

    public int getClienteDni() {
        return clienteDni;
    }

    public void setClienteDni(int clienteDni) {
        this.clienteDni = clienteDni;
    }

    public String getPlato() {
        return plato;
    }

    public void setPlato(String plato) {
        this.plato = plato;
    }
    
    @Override
    protected void cargarDatos(Scanner scanner) {
        this.codigo = scanner.next();
        this.clienteDni = scanner.nextInt();
        this.plato = scanner.next();
    }

    @Override
    protected void imprimir() {
        System.out.println("Codigo: " + this.codigo);
        System.out.println("Cliente DNI: " + this.clienteDni);
        System.out.println("Plato: " + this.plato);
    }
}
